pkgs/os-specific/darwin/apple-sdk-13/default.nix

```
@@ -0,0 +1,133 @@
+{ stdenvNoCC, fetchurl, newScope, lib, pkgs
+, stdenv, overrideCC
+, xar, cpio, python3, pbzx }:
+
+let
+  mkSusDerivation = args: stdenvNoCC.mkDerivation (args // {
+    dontBuild = true;
+    darwinDontCodeSign = true;
+
+    nativeBuildInputs = [ cpio pbzx ];
+
+    outputs = [ "out" ];
+
+    unpackPhase = ''
+      pbzx $src | cpio -idm
+    '';
+
+    passthru = {
+      inherit (args) version;
+    };
+  });
+
+  MacOSX-SDK = mkSusDerivation {
+    pname = "MacOSX-SDK";
+    version = "13.3.0";
+
+    # https://swscan.apple.com/content/catalogs/others/index-13.merged-1.sucatalog
+    # NOTE: In addition to CLTools_macOSNMOS_SDK.pkg, this also contains:
+    # - CLTools_Executables.pkg
+    # - CLTools_macOSLMOS_SDK.pkg
+    # - CLTools_macOS_SDK.pkg
+    # - CLTools_SwiftBackDeploy.pkg
+    src = fetchurl {
+      url = "https://swcdn.apple.com/content/downloads/38/61/032-64167-A_F8LL7XSTW6/k3kg0uip4kxd3qupgy6y8fzp27mnxdpt6y/CLTools_macOSNMOS_SDK.pkg";
+      hash = "sha256-/Wb2uVOutiz5Y19LQjzu6YWi0lQFxTBjBuSRArJrtJs=";
+    };
+
+    installPhase = ''
+      mv Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk $out
+    '';
+  };
+
+  CLTools_Executables = mkSusDerivation {
+    pname = "CLTools_Executables";
+    version = "13.3.0";
+
+    # https://swscan.apple.com/content/catalogs/others/index-13.merged-1.sucatalog
+    src = fetchurl {
+      url = "https://swcdn.apple.com/content/downloads/38/61/032-64167-A_F8LL7XSTW6/k3kg0uip4kxd3qupgy6y8fzp27mnxdpt6y/CLTools_Executables.pkg";
+      hash = "sha256-XkP7RpegWmo3uHG35/imevV+06EU/VE6cNrBEaQThSk=";
+    };
+
+    installPhase = ''
+      mv Library/Developer/CommandLineTools $out
+    '';
+  };
+
+  mkStdenv = stdenv:
+    let
+      cc = stdenv.cc.override {
+        bintools = stdenv.cc.bintools.override { libc = packages.Libsystem; };
+        libc = packages.Libsystem;
+      };
+    in
+    if stdenv.isAarch64 then stdenv
+    else
+      (overrideCC stdenv cc).override {
+        targetPlatform = stdenv.targetPlatform // {
+          darwinMinVersion = "10.12";
+          darwinSdkVersion = "13.3";
+        };
+      };
```

```suggestion
    (overrideCC stdenv cc).override {
      targetPlatform = stdenv.targetPlatform // {
        darwinMinVersion = "10.12";
        darwinSdkVersion = "13.3";
      };
    };
```

Assuming the default SDK will remain 11.0, this should be made unconditional. I’ve made the change locally to test, but I think MoltenVK is failing to build on aarch64-darwin because it would be getting an 11.0 SDK `stdenv` with 13.3 SDK frameworks.

---

> because it would be getting an 11.0 SDK `stdenv` with 13.3 SDK frameworks.

Perhaps something like `darwin.apple_sdk_13_3.callPackage`?

---

I missed the change in default SDK. I reverted that part, and the number of rebuilds are much lower. That should make troubleshooting the MoltenVK failure much easier.

---

It’s not picking up the 13.3 SDK `xcbuild`. 🤔

---

> > because it would be getting an 11.0 SDK `stdenv` with 13.3 SDK frameworks.
> 
> Perhaps something like `darwin.apple_sdk_13_3.callPackage`?

`mkStdEnv` is used to define the 13.3 SDK `stdenv`s provided by `darwin.apple_sdk_13_3.callPackage`, so it can’t use the 13.3 SDK `callPackage` without causing an infinite recursion.

The 11.0 SDK definition should also be made unconditional. When I added support for x86_64-darwin, I wanted to be very conservative about rebuilds, but it’s probably not necessary.

---

Made those changes for the 13.3 SDK, but I can make them for the 11.0 SDK if you feel it wouldn't be out of scope.

Also added some general questions I had as TODOs:

https://github.com/NixOS/nixpkgs/pull/229210/commits/a80d48e2d5d137d001b0673e726081a4f4244417#diff-5b60d4457dd97671b469283ed86615b062fc1eeb634ba9331a8cd1e7a1d46366R67-R77